phpunit/functional/Glpi/Form/AccessControl/FormAccessControlManagerTest.php

```diff
@@ -259,6 +306,72 @@ private function getFormAccessibleOnlyToTechUser(): Form
         );
     }
 
+    private function getActiveFormWithoutAccessControls(): Form
+    {
+        return $this->createForm((new FormBuilder())->setIsActive(true));
+    }
+
+    private function getInactiveFormWithoutAccessControls(): Form
+    {
+        return $this->createForm((new FormBuilder())->setIsActive(false));
+    }
+
+    private function getActiveFormWithActiveAccessControls(): Form
+    {
+        return $this->createForm(
+            (new FormBuilder())
+                ->setIsActive(true)
+                ->addAccessControl(DirectAccess::class, new DirectAccessConfig(
+                    token: 'my_token',
+                ))
+        );
+    }
+
+    private function getInactiveFormWithActiveAccessControls(): Form
+    {
+        return $this->createForm(
+            (new FormBuilder())
+                ->setIsActive(false)
+                ->addAccessControl(DirectAccess::class, new DirectAccessConfig(
+                    token: 'my_token',
+                ))
+        );
+    }
+
+    private function getActiveFormWithInactiveAccessControls(): Form
+    {
+        $form = $this->createForm(
+            (new FormBuilder())
+                ->setIsActive(true)
+                ->addAccessControl(DirectAccess::class, new DirectAccessConfig(
+                    token: 'my_token',
+                ))
+        );
+
+        $control = $this->getAccessControl($form, DirectAccess::class);
+        $this->updateItem($control::class, $control->getID(), ['is_active' => 0]);
+        $form->getFromDB($form->getID());
+
+        return $form;
+    }
+
+    private function getInactiveFormWithInactiveAccessControls(): Form
+    {
+        $form = $this->createForm(
+            (new FormBuilder())
+                ->setIsActive(false)
+                ->addAccessControl(DirectAccess::class, new DirectAccessConfig(
+                    token: 'my_token',
+                ))
+        );
+
+        $control = $this->getAccessControl($form, DirectAccess::class);
+        $this->updateItem($control::class, $control->getID(), ['is_active' => 0]);
+        $form->getFromDB($form->getID());
+
+        return $form;
+    }
```

Comment:
Each of these methods is used only once. Their code should be placed in the corresponding test itself to improve each test case readability.

```php
    public function testGetWarningForInactiveFormWithoutAccessControlPolicies(): void
    {
        $this->checkGetWarnings($this->getInactiveFormWithoutAccessControls(), [
            'This form is not visible to anyone because it is not active.',
            'This form will not be visible to any users as there are currently no active access policies.',
        ]);
    }

    // ....

    private function getInactiveFormWithoutAccessControls(): Form
    {
        return $this->createForm((new FormBuilder())->setIsActive(false));
    }
```

VS

```php
    public function testGetWarningForInactiveFormWithoutAccessControlPolicies(): void
    {
        $form = $this->createForm((new FormBuilder())->setIsActive(false));

        $this->checkGetWarnings($form, [
            'This form is not visible to anyone because it is not active.',
            'This form will not be visible to any users as there are currently no active access policies.',
        ]);
    }
```

Reply:
OK